### PR TITLE
EqDec instance for option

### PR DESCRIPTION
**Pull request by yaitskov**
- [x] Added / updated **test-suite**.

### CODE REVIEW

**Comment by andres-erbsen** [comment]
@coqbot run full ci

**Comment by andres-erbsen** [comment]
@coqbot merge now

File: stdlib/theories/Classes/EquivDec.v [hunk 1]
```diff
@@ -157,3 +157,22 @@ Program Instance list_eqdec `(eqa : EqDec A eq) : EqDec (list A) eq :=
 
   Solve Obligations with unfold equiv, complement in * ;
     program_simpl ; intuition (discriminate || eauto).
+
+#[global]
```
[the remark this answers is not in the document]
Comment (yaitskov) [reply stdlib/theories/Classes/EquivDec.v:161]: a few dozen lines above `Instance list_eqdec` is marked as global. I think these instances must be in sync. That's why I used the pragma for option instance.

[the remark this answers is not in the document]
Comment (ppedrot) [reply stdlib/theories/Classes/EquivDec.v:161]: The fact that we use global above is a historical accident. I personally believe we should refrain from adding global instances in newly written code.

[the remark this answers is not in the document]
Comment (SkySkimmer) [reply stdlib/theories/Classes/EquivDec.v:161]: >FTR I'm no stdlib maintainer but I'm quite convinced that global instances are very anti-modular.

I don't think I'm this absolute about it.

[the remark this answers is not in the document]
Comment (yaitskov) [reply stdlib/theories/Classes/EquivDec.v]: I set the scope to export.

[the remark this answers is not in the document]
Comment (andres-erbsen) [reply stdlib/theories/Classes/EquivDec.v]: I don't feel strongly one way or another. (There are clear issues with global, but less experience with export.)

**Comment by andres-erbsen** [comment]
@coqbot merge now

If the CI passed without this patch, and passed with global, it ought to pass with export. (If there was a "merge when CI passes" option, I'd choose that.)

**Comment by andres-erbsen** [comment]
@coqbot merge now

**Comment by andres-erbsen** [comment]
@coqbot merge now

**Review by andres-erbsen — APPROVED** [review]

**Comment by andres-erbsen** [comment]
@coqbot run full ci

changed my mind after being told off by the bot twice, it's late here...

**Comment by andres-erbsen** [comment]
@coqtop merge now

**Comment by andres-erbsen** [comment]
@coqbot merge now

**Pull request merged**